libs/libxx/mbstate_t.patch

```diff
@@ -0,0 +1,17 @@
+--- libs/libxx/libcxx/include/__mbstate_t.h	2023-11-28 09:52:28.000000000 +0100
```

Comment:
@xiaoxiang781216 so I created a patch file that simply change the order of the if checks, causing it to enter the case we wish. I tested on an docker locally (linux) and it works, we just need to update the docker image to have gcc 12 at least. Let me know what you think.

Reply:
Is there a procedure somewhere to update the docker image? I mean i can modify the dockerfile but i do not think i will have the right push to the registry

Reply:
The image is automatically push to the server once Dockfile is merged to git.

Reply:
ah ok, i guess i'll make a separate MR for the docker image. Let's merge that first that way we are sure nothing is broken and then we'll see if it resolves our issue here. Sounds good?

Reply:
@xiaoxiang781216 here is the update to the docker file: https://github.com/apache/nuttx/pull/11333

Reply:
Merged, please rebase your patch.

Reply:
should be done at the merging:
https://github.com/apache/nuttx/actions/runs/7116238621/job/19374199719

Reply:
@xiaoxiang781216 https://github.com/apache/nuttx/actions/runs/7116873366/job/19376297186 there is the one for master right after that seemed to have failed

Reply:
could you provide a patch?

Reply:
@xiaoxiang781216 well i guess we cannot use it as they are not verified in the github marketplace

Reply:
But, the patch pass ci

Reply:
Yes we need to change the rules somewhere in the settings to allow us to use unverified actions in our workflow. Unless we really do not want that then I'd suggest to drop the commit.

Reply:
The patch ci passed because I am guessing it used the old docker image to run the tests

Reply:
> Yes we need to change the rules somewhere in the settings to allow us to use unverified actions in our workflow. Unless we really do not want that then I'd suggest to drop the commit.

this behavior is probably controlled by Apache infra, so it's not that easy to change. 
Can't we just port this action to our repo (it's MIT license) with all the credits to the author and a note in LICENSE: https://github.com/jlumbroso/free-disk-space/blob/main/action.yml

Reply:
@raiden00pl sounds good to me. I'll make a PR porting their action to ours. I'll ping you once i have something working.

Reply:
@raiden00pl here you go https://github.com/apache/nuttx/pull/11354

Reply:
let's wait this pr pass ci:
https://github.com/apache/nuttx/pull/11346
after it get merged, the new docker image should contain your change.

Reply:
> let's wait this pr pass ci: #11346 after it get merged, the new docker image should contain your change.

@xiaoxiang781216 i think it is too late i think it has already been merged

Reply:
typo error, the patch is https://github.com/apache/nuttx/pull/11345